### DISCUSSION TITLE
Custom docker image build takes a long time and then gets stuck

### DISCUSSION THREAD

**Discussion by vijayreddybomma** [apache/superset#25418] [Q&A / Help]
When attempting to create a custom docker image of Apache Superset inside a Docker locally, it gets stuck at 94% during the build process and it's not  progressing further.

Can I know on average how much time it'll take to build image?

**Comment by sfirke** [comment]
Hello - this is hard to answer exactly as the image build time would depend on the power of the machine building it and what modifications you've added.  I think past image builds were taking me on the order of 10-15 minutes for vanilla Superset.

I'm going to convert this to a GitHub Discussion since it's a question, not a bug report.

**Reply by vijayreddybomma** [reply]
```
=> ERROR [superset-node  8/10] RUN npm ci                                                                                                                                                                 1655.2s
------                                                                                                                                                                                                             
 > [superset-node  8/10] RUN npm ci:                                                                                                                                                                               
1655.2 Killed                                                                                                                                                                                                      
------
Dockerfile:40
--------------------
  38 |     COPY superset-frontend/package*.json ./
  39 |     RUN npm install
  40 | >>> RUN npm ci
  41 |     
  42 |     COPY ./superset-frontend .
--------------------
ERROR: failed to solve: process "/bin/sh -c npm ci" did not complete successfully: exit code: 137

```


This is the error I'm getting after build process took at least 1 hour

**Reply by sfirke** [reply]
Can you try just cloning the repo (set tag to 3.0.0 if you want to make sure you're using a stable version) and then building that image?  If that works then you know it's an issue with your customization code, and either way you'd get a benchmark for what the build time should be without your customizations.

**Reply by vijayreddybomma** [reply]
```
Dockerfile:53
--------------------
  51 |     
  52 |     # This seems to be the most expensive step
  53 | >>> RUN npm run ${BUILD_CMD}
  54 |     
  55 |     ######################################################################
--------------------
ERROR: failed to solve: process "/bin/sh -c npm run ${BUILD_CMD}" did not complete successfully: exit code: 1
```


As u said, I tried to build an image(tag 3.0.0), without any modifications in the code but encountering with above error!


@sfirke

**Reply by vijayreddybomma** [reply]
also when I run ``` npm run build ``` locally its just shutting down my computer,  Can u say me,  if I need to change any docker configuration settings in the resources section?

**Reply by vijayreddybomma** [reply]
![Screenshot from 2023-09-27 14-17-35](https://github.com/apache/superset/assets/66731645/ddc7afac-7a5f-4012-8a37-82d3d637932e)

**Comment by chetanbothra** [comment]
Any solution for this ? i m getting the same error

**Comment by sadatbs** [comment]
Is there any update on this issue? Facing the same problem.

**Reply by sadatbs** [reply]
For anyone facing the same issue out there. I am building the superset image with docker in windows system. The issue was resolved by changing the prettier.config.js file in the superset-frontend diectory as follows -
module.exports = {
  singleQuote: true,
  trailingComma: 'all',
  arrowParens: 'avoid',
  endOfLine: 'auto',
};

**Reply by dmitribogonos** [reply]
> For anyone facing the same issue out there. I am building the superset image with docker in windows system. The issue was resolved by changing the prettier.config.js file in the superset-frontend diectory as follows - module.exports = { singleQuote: true, trailingComma: 'all', arrowParens: 'avoid', endOfLine: 'auto', };

Thanks a lot, seems to work!
Unfortunately results in a new issue later:

```bash
.
.
.
 => CACHED [superset-worker-beat superset-node  9/12] COPY superset/translations /app/superset/translations                          0.0s
 => ERROR [superset-worker-beat superset-node 10/12] RUN npm run build-translation                                                   0.7s
------
 > [superset-worker-beat superset-node 10/12] RUN npm run build-translation:
0.674
0.674 > superset@0.0.0-dev build-translation
0.674 > scripts/po2json.sh
0.674
0.679 sh: 1: scripts/po2json.sh: not found
------
failed to solve: process "/bin/sh -c npm run build-translation" did not complete successfully: exit code: 127
```
didn't find any solution yet, may be worth opening a new issue here

**Reply by cuongnm** [reply]
> > For anyone facing the same issue out there. I am building the superset image with docker in windows system. The issue was resolved by changing the prettier.config.js file in the superset-frontend diectory as follows - module.exports = { singleQuote: true, trailingComma: 'all', arrowParens: 'avoid', endOfLine: 'auto', };
> 
> Thanks a lot, seems to work! Unfortunately results in a new issue later:
> 
> ```shell
> .
> .
> .
>  => CACHED [superset-worker-beat superset-node  9/12] COPY superset/translations /app/superset/translations                          0.0s
>  => ERROR [superset-worker-beat superset-node 10/12] RUN npm run build-translation                                                   0.7s
> ------
>  > [superset-worker-beat superset-node 10/12] RUN npm run build-translation:
> 0.674
> 0.674 > superset@0.0.0-dev build-translation
> 0.674 > scripts/po2json.sh
> 0.674
> 0.679 sh: 1: scripts/po2json.sh: not found
> ------
> failed to solve: process "/bin/sh -c npm run build-translation" did not complete successfully: exit code: 127
> ```
> 
> didn't find any solution yet, may be worth opening a new issue here

Open file po2json.sh (with VSCode), change "End of Line Sequence" from CRLF to LF.

**Comment by olka** [comment]
Same here on MacBook pro m1 with qemu/colima 
Getting stuck and subsequently killed on `npm run build` step

**Comment by vmaksimenko** [comment]
I experienced the same problem running the Superset docker compose on Windows 11 WSL2. I waited a long time, and when npm run build finished after 40 minutes, with 30K+ errors about prettier/prettier: Delete ␍. As I understand, it occurs due to Windows and Linux end-of-line differences when copying files in a Docker image. Running the following commands on the project level helped me. After the fix build runs faster and with success.
```
git config core.autocrlf false
git rm --cached -r .
git reset --hard
```
